instance/models/mixins/openedx_database.py

```diff
@@ -234,6 +236,19 @@ def set_field_defaults(self):
         if not self.mongo_user:
             self.mongo_user = get_random_string(length=16, allowed_chars=string.ascii_lowercase)
             self.mongo_pass = get_random_string(length=32)
+        if not self.rabbitmq_vhost:
+            def new_rabbitmq_user():
```

Comment:
Static method sounds good. Since you're subclassing ValidateModelMixin for the RabbitMQUser object, you could also generate the username and password on save with `set_field_defaults`.